hs-abci-extra/src/Network/ABCI/Middleware/RequestLogger.hs

```diff
@@ -1,23 +1,66 @@
     ) where
 
-import           Control.Monad.IO.Class                 (MonadIO, liftIO)
-import           Data.Aeson                             (FromJSON (..),
-                                                         ToJSON (..),
-                                                         genericParseJSON,
-                                                         genericToJSON)
-import           Data.Aeson.Casing                      (aesonPrefix, camelCase)
+import           Control.Monad.IO.Class              (MonadIO, liftIO)
+import           Data.Aeson                          (FromJSON (..),
+                                                      ToJSON (..),
+                                                      genericParseJSON,
+                                                      genericToJSON)
+import           Data.Aeson.Casing                   (aesonPrefix, camelCase)
 import           Katip
+import           System.IO                           (stdout)
 
 
-import           Network.ABCI.Types.App                 (App (..), Middleware)
-import           Network.ABCI.Types.Messages.FieldTypes
+import           Network.ABCI.Types.App              (App (..), Middleware)
+import           Network.ABCI.Types.Messages.Request (Request)
 
--- | Production request logger middleware for ABCI requests
-logStdout :: (MonadIO m) => LogEnv -> Namespace -> Middleware m
-logStdout env ns (App app) = App $ \ req -> undefined
-  -- liftIO (runKatipContextT env req ns $ $(logTM) InfoS "Request Received")
-  -- >> app req
+---------------------------------------------------------------------------
+-- | Types
+---------------------------------------------------------------------------
+-- | Loggable newtype wrapper
+newtype Loggable a = Loggable a
+instance ToJSON (Loggable (Request t)) where
+  toJSON (Loggable a) = toJSON a
 
+instance ToObject (Loggable (Request t))
+instance LogItem (Loggable (Request t)) where
+  payloadKeys _ _ = AllKeys
+
+---------------------------------------------------------------------------
+-- | mkLogStdout
+---------------------------------------------------------------------------
+-- | Creates a production request katip logger as middleware for ABCI
+-- requests.
+mkLogStdout :: (MonadIO m) => m (Middleware m)
+mkLogStdout = do
+  handleScribe <- liftIO $ mkHandleScribe ColorIfTerminal stdout InfoS V2
+  le <- liftIO (registerScribe "stdout" handleScribe defaultScribeSettings =<< initLogEnv "ABCI" "production")
+  let ns = "Server"
+  pure $ mkRequestLogger le ns
+
+---------------------------------------------------------------------------
+-- | mkRequestLogger
+---------------------------------------------------------------------------
+-- | Request logger middleware for ABCI requests with custom Katip LogEnv
+-- and Namespace.
+mkRequestLogger :: (MonadIO m) => LogEnv -> Namespace -> Middleware m
+mkRequestLogger le ns (App app) = App $ \ req -> do
+  runKatipContextT le () ns $ logRequest req
+  app req
```

Comment:
shouldn't we also log response?

Reply:
Yeah we could, I was following the [wai-request-logger](http://hackage.haskell.org/package/wai-extra-3.0.27/docs/Network-Wai-Middleware-RequestLogger.html). I think the response logger would be a separate piece of middle ware

Reply:
good point, and we can add it later.